app/src/main/java/agency/five/codebase/android/movieapp/data/di/DataModule.kt

```diff
@@ -1,18 +1,26 @@
 package agency.five.codebase.android.movieapp.data.di
 
-import agency.five.codebase.android.movieapp.data.repository.FakeMovieRepository
+
 import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.data.repository.MovieRepositoryImpl
 import kotlinx.coroutines.Dispatchers
 import org.koin.core.qualifier.named
-
-
 import org.koin.dsl.module
 
 val dataModule = module {
+
     single<MovieRepository>(named("dispatcherIo")) {
-        FakeMovieRepository(ioDispatcher = Dispatchers.IO)
+        MovieRepositoryImpl(
+            movieService = get(),
+            movieDao = get(),
+            bgDispatcher = Dispatchers.IO
+        )
     }
     single<MovieRepository>(named("dispatcherMain")) {
-        FakeMovieRepository(ioDispatcher = Dispatchers.Main)
+        MovieRepositoryImpl(
+            movieService = get(),
+            movieDao = get(),
+            bgDispatcher = Dispatchers.Main
+        )
```

Comment:
why do you have two repositories?

why do you need a dispatcher main in the repository?